File: server/models/EventComment.js
const mongoose = require('mongoose');

const EventCommentSchema = new mongoose.Schema({
  eventId: { type: mongoose.Schema.Types.ObjectId, required: true, ref: 'Event' },
  content: { type: String, required: true },
  date: { type: Date, default: Date.now },
});

module.exports = mongoose.model('EventComment', EventCommentSchema);

File: lib/models/event_comment.dart
part of 'models.dart';
class EventComment {
  final String? id;
  final int eventId;
  final String content;
  final DateTime date;

  EventComment({
    this.id,
    required this.eventId,
    required this.content,
    DateTime? date,
  }) : date = date ?? DateTime.now();

  factory EventComment.fromMap(Map<String, dynamic> map) => EventComment(
        id: map['id'] as String?,
        eventId: map['eventId'] is int
            ? map['eventId'] as int
            : int.parse(map['eventId'].toString()),
        content: map['content'] as String,
        date: _parseDate(map['date']),
      );

  Map<String, dynamic> toMap() => {
        if (id != null) 'id': id,
        'eventId': eventId,
        'content': content,
        'date': date.toIso8601String(),
      };

  factory EventComment.fromJson(Map<String, dynamic> json) =>
      EventComment.fromMap(json);
  Map<String, dynamic> toJson() => toMap();
}
